symphony-ws/src/main/java/se/havochvatten/symphony/service/ReportService.java

```diff
@@ -420,4 +415,290 @@ static void writeComparisonRows(Locale locale, SymphonyBandDto[] meta, int[] fea
+                cmpResult.cumulativeTotal = cmpResults[i].cumulativeTotal;
+                return cmpResult;
+
+            }).toArray(ComparisonResultExport[]::new);
+
+        return result;
+    }
+
+    private OdfTable createSheet(OdfSpreadsheetDocument doc, int rows, int cols, String title) {
+        OdfTable newSheet = OdfTable.newTable(doc, rows, Math.min(cols, 5));
+
+        newSheet.getCellRangeByPosition(2, 1, 5, 1).merge();
+        OdfTableCell titleCell = newSheet.getCellByPosition(2, 1);
+        titleCell.setStringValue(title);
+        titleCell.getOdfElement().setStyleName("cmpName");
+
+        newSheet.getColumnByIndex(0).setWidth(5);
+        newSheet.getRowByIndex(1).setHeight(14, false);
+
+        return newSheet;
+    }
+
+    public byte[] generateMultiComparisonAsODS(
+        CompoundComparison comparison, String preferredLanguage, String localisedHeading, boolean excludeZeroes) throws Exception {
+
+        String title = localisedHeading + " \"" + comparison.getName() + "\"";
+
+        Map<Integer, String>
+            ecoTitles = metaDataService.getComponentTitles(
+                comparison.getBaseline().getId(),
+                LayerType.ECOSYSTEM, preferredLanguage),
+            pressureTitles = metaDataService.getComponentTitles(
+                comparison.getBaseline().getId(),
+                LayerType.PRESSURE, preferredLanguage);
+
+        // The ODF template file 'template.ods' that is packaged with the application is prepared
+        // with the following named office styles (in "styles.xml") :
+        // "cmpName", "calcName", "ecoHeader", "pressureHeader", "value", "totalHE", "totalHP",
+        // "totalE", "totalP", "totalC", "resultSep"
+
+        OdfPackage odfPackage = loadPackage(
+            getClass().getClassLoader().getResourceAsStream("document/template.ods")
+        );
+
+        OdfSpreadsheetDocument templateDocument = (OdfSpreadsheetDocument) odfPackage.getRootDocument();
+
+        ComparisonResult[] cmpResults =
+            excludeZeroes ?
+                comparison.getResult().values().stream()
+                    .filter(cmp -> cmp.cumulativeTotal != 0)
+                    .toArray(ComparisonResult[]::new) :
+            comparison.getResult().values().toArray(new ComparisonResult[0]);
+
+        OdfTable totalSheet = createSheet(
+            templateDocument,
+            cmpResults.length * 7 + ODF_TITLE_ROWS_TOTAL, 3,
+            title);
+
+        totalSheet.setTableName(comparison.getName() + " - Total");
+        totalSheet.getColumnByIndex(1).setWidth(ODF_TITLE_COLWIDTH);
+
+        // Semantically consistent index variables.
+        int e, p, j, cmpIndex = 0;
+
+        for (ComparisonResult cmp : cmpResults) {
+            OdfTable nextSheet = createSheet(
+                templateDocument,
+                cmp.includedEcosystems.length + ODF_TITLE_ROWS, cmp.includedPressures.length + 1,
+                title);
+            OdfTableCell nextCell;
```

Comment:
No, it isn't.
As you can see nextCell is reused throughout the loop (to avoid instantiating new objects on the stack for every cell that we need to manipulate). I feel making this a mutable variable is best from the standpoint of the JVM.

Reply:
Yes I meant only to instantiating once but on the same row :) But I also think it is more clear that we re-use it as it is now.

Reply:
I recognise that I misunderstood you slightly!
Good we're on the same page about utilizing this pattern. 🙂